packages/shared/src/parsers.ts

```diff
@@ -20,6 +20,13 @@ const errorMap: ZodErrorMap = (_issue, context) => ({
   message: `received: ${safeStringify(context.data)}`,
 });
 
+export const hexStringWithMaxByteSize = (maxByteSize: number) =>
+  z
+    .string()
+    .refine((val) => /^0x[0-9a-f]+$/.test(val))
```

Comment:
Do we need to care about case-sensitivity here? 
I know that in the ethereum ecosystem, the wallet addresses are many times represented as `/^0x[0-9a-fA-F]+$/`. Users use all sorts of tools to copy these addresses and sometimes the result is, they have a checksummed address in their clipboard.
I guess it depends on what we expect to be passed as a value for the ccm but just wanted to raise the awareness on this one.

Reply:
good point! im actually not sure how the backend handles this - worth a check if it accepts uppercase letters

Reply:
yep sounds good - can we reuse/adjust the existing `hexString`?

Reply:
updated it to reuse `hexString` which actually is case insensitive